dropwizard-jackson/src/main/java/io/dropwizard/jackson/SafeDurationDeserializer.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2013 FasterXML.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package io.dropwizard.jackson;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.JsonTokenId;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.deser.std.StdScalarDeserializer;
+import com.fasterxml.jackson.datatype.jsr310.DecimalUtils;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.DateTimeException;
+import java.time.Duration;
+import java.util.Arrays;
+
+/**
+ * This class is taken from the jackson-jsr310 module and modified to reject big decimal
+ * values. Unfortunatelly, `DurationDeserializer` has a private constructor and
+ * can't be extended, so we have to copy.
+ */
+public class SafeDurationDeserializer extends StdScalarDeserializer<Duration> {
```

Comment:
While you may not be able to extend `DurationDeserializer`, have you seen how what the implementation would look like if you delegated to `DurationDeserializer.INSTANCE`, after bounds checking? I feel like the code may be shorter and less prone to maintenance.

Reply:
I haven't, but that makes a hell of a lot sense to me. I had presumed `DurationDeserializer.INSTANCE` is package protected.